src/components/Tree/Tree.tsx

```diff
@@ -15,6 +15,7 @@ export interface TreeFlatListItem {
     label?: string;
     value?: string;
     actions?: React.ReactNode[];
+    badge?: ReactElement<IconProps>;
```

Comment:
Would we be able to restrict a bit more what we put into the Badge? Because its called Badge, i think it would make sense to restrict to Icons/Badge, and also what you would need to fulfill your requirements.